OpenSim/Common/Component.cpp

```diff
@@ -192,6 +192,48 @@ void Component::finalizeFromProperties()
 
     markPropertiesAsSubcomponents();
     componentsFinalizeFromProperties();
+
+    auto subs = getImmediateSubcomponents();
+    std::set<std::string> names{};
+    std::string name{};
+
+    // increment the count of duplicates to use as a unique suffix
+    int count{ 0 };
+    // temp variable to hold the unique name used to rename a duplicate
+    std::string uniqueName{};
+
+    for (auto& sub : subs) {
+        name = sub->getName();
+
+        // reset duplicate count and search name
+        count = 0;
+        uniqueName = name;
+
+        // while the name is still not unique keep incrementing the count
+        while (names.find(uniqueName) != names.cend()) {
+            // In the future this should become an Exception 
+            //OPENSIM_THROW(SubcomponentsWithDuplicateName, getName(), searchName);
+            // for now, rename the duplicately named subcomponent 
+            // but first test the uniqueness of the name (the while condition)
+            uniqueName = name + "_" + std::to_string(count++);
+        }
+
+        if (count > 0) { // if a duplicate
+            // rename the the subcomponent with its verified unique name
+            Component* mutableSub = const_cast<Component *>(sub.get());
```

Comment:
Consider adding a `std::vector<SimTK::ReferencePtr<Component>> getImmediateSubcomponents() const;`

Reply:
I think I will do that in a future PR and also use it to avoid cycling through the three internal lists. I was reluctant to add another accessor for this one use, but now I think I would use it.